riscv/src/compiler.rs

```diff
@@ -357,17 +458,37 @@ fn substitute_symbols_with_values(
     statements
 }
 
-fn risv_machine(preamble: &str, main: &str) -> String {
+fn risv_machine(
```

Comment:
```suggestion
fn riscv_machine(
```
Since you're touching this line anyway.